pkg/operator/terminationobserver/termination_observer.go

```diff
@@ -0,0 +1,249 @@
+package terminationobserver
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+var (
+	controllerWorkQueueKey = "key"
+
+	// terminationEventReasons lists all events that are observed when an API server is shutting down gracefully.
```

Comment:
do we have these for the openshift-apiserver or no?

Reply:
@sttts ^ i think every API server has them, if this proof to be useful, we can run this in openshift-apiserver operator as well and I will librarize it ;-)